### DISCUSSION TITLE
Does  `nvim-jdtls` support workspace symbols other than `class` (like `method`, `field`, `constructor`, etc?

### DISCUSSION THREAD

**Discussion by ecerulm** [mfussenegger/nvim-jdtls#676] [Q&A] [answered]
I know that for document symbols it will find `method`, `field`, etc. 
```
:lua vim.lsp.buf.document_symbol({}) 
```

but when I use `vim.lsp.buf.workspace_symbol` I can't find anything that is not a `class`:  

```
:lua vim.lsp.buf.workspace_symbol("tokens",{})
```

So is this because I misconfigure something, or it's just not supported by `nvim-jdtls` or maybe it's missing from `jdtls`/`eclipse.jdt.ls` altogether?

**Comment by rogeriofrsouza** [comment]
Seems like this is not supported by `jdtls` itself.
As per this issue https://github.com/redhat-developer/vscode-java/issues/1712#issuecomment-1059524516, that was a design choice because of performance reasons.

**Reply by ecerulm** [reply, marked as answer]
Perfect, it seems that `jdtls` support at least returning java methods in the workspace in addition to just class, via `includeSourceMethodsDeclarations` config.

So now I'm using


```
local config = {
	cmd = { "jdtls" },
	root_dir = vim.fs.dirname(vim.fs.find({ "gradlew", ".git", "mvnw" }, { upward = true })[1]),
	settings = {
		java = {
			configuration = {
...
			},
			symbols = {
				includeSourceMethodDeclarations = true,
			},
		},
	},
}
require("jdtls").start_or_attach(config)

```

And now when I run `:Telescope lsp_dynamic_workspace_symbols` I can find `method` and `class`
 
<img width="708" alt="image" src="https://github.com/user-attachments/assets/4caa1bfc-d04e-4479-8998-ebebd3d10cfb">
